### DISCUSSION TITLE
NumberFormatException in nullable Int

### DISCUSSION THREAD

**Discussion by PeterAttardo** [pdvrieze/xmlutil#133] [Q&A] [answered]
I am currently using the library to parse the XML responses of an API I do not control. One of the fields in this API is an attribute that represents an integer, and may or may not have a value. When it does not have a value, it is represented as an empty string in the delivered XML. These instances are currently throwing a `NumberFormatException` because it appears that an empty string is not read as `null` the way a missing attribute would be, but is also not parseable as an `Int`. Is there a way to tell the parser that any empty string should be treated as `null` or that any unparseable value should be treated as `null`? Either one would solve my usecase.

Thanks

**Comment by pdvrieze** [comment, marked as answer]
The current implementation does not have support for recovering from errors in the parsing of contents (including number formats). There are 3 workarounds:

- pre-filter the xml to "tidy" it up
- Use a custom serializer that parses the content manually (in the case of xml you can just use readString).
- Use a string property for the serialized type and handle the property value in the type. (You can combine this with a serializer that uses this type as the input for the type you actually want)

**Reply by PeterAttardo** [reply]
I think in this case setting everything as a string value and handling the parsing myself is probably the simplest workaround.

Thanks
